### PR TITLE
Guard against overflow for dynamic interval overlap

### PR DESCRIPTION
**Pull request by whimxiqal**
The dynamic interval `overlaps` method contains addition code to the boundaries regardless of whether those values would overflow. This can cause wrong results at, at the very least, compiler errors depending on the compiler options.

These changes guard against such addition/subtraction and add tests with additional compiler options to error on integer overflow.

I've also adjusted the workflow, CMake, and README files to streamline testing a bit.

### CODE REVIEW

**Comment by whimxiqal** [comment]
Fixes https://github.com/5cript/interval-tree/issues/61

**Comment by 5cript** [comment]
Thank you for the contribution
I will review this as soon as I am able.

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -47,4 +54,9 @@ if (DEFINED ENV{MSYSTEM})
         COMMAND bash -c "ldd $<TARGET_FILE:tree-tests>" | "grep" "clang" | awk "NF == 4 { system(\"${CMAKE_COMMAND} -E copy \" \$3 \" $<TARGET_FILE_DIR:tree-tests>\") }"
         VERBATIM
     )
+endif()
+
+if(CMAKE_C_COMPILER_ID MATCHES "Clang|GNU")
```
Comment (5cript) [inline tests/CMakeLists.txt]: CMAKE_CXX_COMPILER_ID

File: README.md [hunk 2]
```diff
@@ -56,9 +56,15 @@ int main()
...
 
 ## Compile & Run Testing
-Having googletest (find here on github) installed / built is a requirement to run the tests.
 Create a build folder, navigate there, run cmake and build the tree-tests target.
-You might have to adapt the linker line for gtest, if you built it yourself and didn't install it into your system.
+
+```bash
+mkdir build
+cd build
+cmake --build .
```
Comment (5cript) [inline README.md:65]: cmake .. -DINT_TREE_ENABLE_TESTS=on

Configure step is missing.

File: tests/CMakeLists.txt [hunk 3]
```diff
@@ -19,7 +19,14 @@ file(GLOB sources "*.cpp")
 # Add Executable
 add_executable(tree-tests ${sources})
 
-find_package(GTest REQUIRED)
+include(FetchContent)
+FetchContent_Declare(
```
Comment (5cript) [inline tests/CMakeLists.txt:23]: Yes this is easier and I only accept this because its for tests only where it doesnt matter.
FetchContent cannot be used in network isolated contexts such as flatpak and yocto.

Just letting you know.

**Comment by 5cript** [comment]
I did some commits on top:
https://github.com/5cript/interval-tree/compare/whimxiqal:bug/overlap-overflow...5cript:bug/overlap-overflow-review

Can i turn this review upside down? :D
I think the fix you introduced doesnt work. Because it disregards the negative numbers.
I added some tests to cover for that and I think fixed the problem properly.

I can put them on top of your contribution or follow them up afterwards. Or you cherry pick them yourself
Waiting for reply

**Comment by whimxiqal** [comment]
Ah good point with the negative numbers. Cherry-picked!

**Review by 5cript — APPROVED** [review]

**Comment by 5cript** [comment]
Thanks a lot for improving this library!
